src/_pytest/doctest.py

```diff
@@ -108,13 +109,18 @@ def pytest_unconfigure():
     RUNNER_CLASS = None
 
 
-def pytest_collect_file(path, parent):
+def pytest_collect_file(
+    path: py.path.local, parent
+) -> Optional[Union["DoctestModule", "DoctestTextfile"]]:
     config = parent.config
     if path.ext == ".py":
         if config.option.doctestmodules and not _is_setup_py(config, path, parent):
-            return DoctestModule.from_parent(parent, fspath=path)
+            mod = DoctestModule.from_parent(parent, fspath=path)  # type: DoctestModule
+            return mod
     elif _is_doctest(config, path, parent):
-        return DoctestTextfile.from_parent(parent, fspath=path)
+        txt = DoctestTextfile.from_parent(parent, fspath=path)  # type: DoctestTextfile
+        return txt
```

Comment:
Instead of splitting these out so you can add a comment, annotate [the `from_parent` method](https://github.com/bluetech/pytest/blob/49f830d1337b4949ad3684322d78bf6de08376ee/src/_pytest/nodes.py#L148-L150)

Reply:
`from_parent` methods can't be properly typed as-is, there is discussion on this in #6387, for now I just work around it.